### PR TITLE
contrib/kind: enable XDP_TX from pod veth

### PR DESCRIPTION
**Pull request by lmb**
Both veth in a pair require an XDP program installed for XDP_TX
to work. Since the host side veth created by kind doesn't have
an XDP program attached we can't run any tests in CI that require
XDP_TX.

The workaround itself is just an ip link set and ethtool away,
the problem is figuring out which interfaces we need to do the
magic to.

Use the approach used by kind-network-plugins and create our own
docker network with a specific name for the bridge device. We
can then iterate all children of the bridge and do our fixups.

We tell kind to use our own network by setting the (undocumented?)
KIND_EXPERIMENTAL_DOCKER_NETWORK environment variable.

See https://github.com/aojea/kind-networking-plugins

Required for #24151.

```release-note
Enable testing of BPF programs requiring XDP_TX in CI
```

### CODE REVIEW

**Comment by brb** [comment]
Please rebase your PR against the latest master. Also, you might want to update `.github/workflows/conformance-datapath.yaml` instead of the v1.13 equivalent.

File: .github/workflows/conformance-datapath-v1.13.yaml [hunk 1]
```diff
@@ -286,15 +286,15 @@ jobs:
           provision: 'false'
           cmd: |
             cd /host/
-            ./contrib/scripts/kind.sh "" 3 "" "" "${{ matrix.kube-proxy }}" "dual"
+            ./contrib/scripts/kind.sh --xdp "" 3 "" "" "${{ matrix.kube-proxy }}" "dual"
```
Comment (lmb) [inline .github/workflows/conformance-datapath-v1.13.yaml:289]: Not sure if changing it for 1.13 is sensible?

Comment (brb) [reply .github/workflows/conformance-datapath-v1.13.yaml:289]: https://github.com/cilium/cilium/pull/24470 got merged, so the master failure should be fixed. We can change the v1.13 later on.

Anyway, I'd suggest to add `xdp: true` to the configuration matrix, and then conditionally to enable it.

File: .github/workflows/conformance-datapath.yaml [hunk 2]
```diff
@@ -389,7 +389,7 @@ jobs:
               --sysdump-output-filename "cilium-sysdump-${{ matrix.name }}-<ts>"
             ./cilium-cli connectivity test --collect-sysdump-on-failure \
               --sysdump-output-filename "cilium-sysdump-${{ matrix.name }}-<ts>"
-            kind delete cluster
```
Comment (nebril) [inline .github/workflows/conformance-datapath.yaml:392]: I wonder if we even need to delete the cluster here. This is the last job in the workflow AFAIU, so we can just skip deletion as node is going to be discarded anyway.

**Review by nebril — APPROVED** [review]
CI changes LGTM, on question left inline.

File: .github/workflows/conformance-datapath-v1.13.yaml (hunk 1 above)
Comment (lmb) [reply .github/workflows/conformance-datapath-v1.13.yaml:289]: Is there a downside to always enabling it on CI? My thinking was that I'd add something like `lb-acceleration: testing-only` to the matrix instead.

File: .github/workflows/conformance-datapath.yaml (hunk 2 above)
Comment (lmb) [reply .github/workflows/conformance-datapath.yaml:392]: I don't understand the workflow well enough to make that call. Would it not affect the `Fetch artifacts` step below it?

Comment (brb) [reply .github/workflows/conformance-datapath.yaml:392]: The `Fetch artifacts` step is going to be executed only if any of the connectivity tests above fails. In that case, the `kind delete` will be bypassed.

File: .github/workflows/conformance-datapath-v1.13.yaml (hunk 1 above)
Comment (brb) [reply .github/workflows/conformance-datapath-v1.13.yaml:289]: I don't see any downside enabling it unconditionally, so yeah I'm fine either way.

**Review by brb — COMMENTED** [review]
@lmb Could you remove the test commit? Once it's removed, I will ACK and mark as `ready-to-merge`.

**Review by brb — APPROVED** [review]
:rocket:

File: .github/workflows/conformance-datapath-v1.13.yaml (hunk 1 above)
Comment (tklauser) [reply .github/workflows/conformance-datapath-v1.13.yaml:289]: This change in 1.13 workflows breaks datapath test in 1.13 backports: https://github.com/cilium/cilium/actions/runs/4544793510

Turns out backporting just this PR to fix it is not enough. Thus, I've sent a partial revert for now: https://github.com/cilium/cilium/pull/24611

Comment (lmb) [reply .github/workflows/conformance-datapath-v1.13.yaml:289]: Sorry, I forgot to back out this change.

**Comment by lmb** [comment]
That's what I had done initially, but it's kind of difficult to figure out whether this would break anybodies workflow. If you want to try you could add `--xdp` to `make kind` and then ask people to try that branch locally?

**Pull request merged**